common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -41,6 +42,9 @@
  */
 public abstract class SingleThreadEventExecutor extends AbstractScheduledEventExecutor {
 
+    protected static final int DEFAULT_MAX_PENDING_TASKS = Math.max(16,
+            SystemPropertyUtil.getInt("io.netty.eventexecutor.maxPendingTasks", 1024 * 1024));
```

Comment:
I think this should be good enough too ... previously it looks like we just did `LinkedBlockingQueue<Runnable>()` which will default to `Integer.MAX_VALUE` capacity.